spec/lib/unique_within_region_validator_spec.rb

```diff
@@ -2,65 +2,46 @@
   describe "#unique_within_region" do
     context "class without STI" do
       let(:case_sensitive_class) do
-        Class.new(User).tap do |c|
-          c.class_eval do
-            validates :name, :unique_within_region => true
-          end
+        Class.new(ApplicationRecord) do
+          self.table_name = "users"
+          validates :name, :unique_within_region => true
         end
       end
 
       let(:case_insensitive_class) do
-        Class.new(User).tap do |c|
-          c.class_eval do
-            validates :name, :unique_within_region => {:match_case => false}
-          end
+        Class.new(ApplicationRecord) do
+          self.table_name = "users"
+          validates :name, :unique_within_region => {:match_case => false}
         end
       end
 
       let(:scoped_class) do
-        Class.new(User).tap do |c|
-          c.class_eval do
-            validates :name, :unique_within_region => {:scope => :email}
-          end
+        Class.new(ApplicationRecord) do
+          self.table_name = "users"
+          validates :name, :unique_within_region => {:scope => :email}
         end
       end
 
       let(:test_name)  { "thename" }
-      let(:test_email) { "thename@example.com" }
 
       let(:in_first_region_id) do
-        FactoryBot.create(
-          :user,
+        case_sensitive_class.create(
```

Comment:
Should these use `create!`?

Reply:
`FactoryBot.create` throws an error for invalid objects just like `Model.create!`
`FactoryBot.create!` is not a valid method

this still makes me double take though.
wish it were closer to rails standards

Reply:
ugh. thanks @bdunne for being patient.
all set